Cron/RetrieveReviews.php

```diff
@@ -2,90 +2,108 @@
 
 namespace Elgentos\Kiyoh\Cron;
 
+use Magento\Framework\Exception\NoSuchEntityException;
 use Magento\Framework\HTTP\Client\Curl;
 use Magento\Framework\Module\Declaration\Converter\Dom;
-use Magento\Framework\View\Element\Template\Context;
+use Magento\Variable\Model\Variable;
 use Magento\Variable\Model\VariableFactory;
 use \Psr\Log\LoggerInterface;
-use Magento\Store\Model\StoreManagerInterface;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Framework\Exception\CronException;
+use Elgentos\Kiyoh\Model\Config;
 
 class RetrieveReviews {
 
     /**
      * @var Curl
      */
-    private $curl;
+    private Curl $curl;
+
     /**
-     * @var StoreManagerInterface
+     * @var LoggerInterface
      */
-    private $storeManager;
+    private LoggerInterface $log;
+
     /**
      * @var Dom
      */
-    private $converter;
+    private Dom $converter;
 
-    /**
-     * @var LoggerInterface
-     */
-    private $log;
     /**
      * @var VariableFactory
      */
-    private $variable;
+    private VariableFactory $variable;
 
-    private $filePath;
+    /**
+     * @var Json
+     */
+    private Json $jsonSerializer;
 
+    /**
+     * @var Config
+     */
+    private Config $config;
 
     public function __construct(
-        Context $context,
-        StoreManagerInterface $storeManager,
         Curl $curl,
-        Dom $converter,
+        Json $jsonSerializer,
         LoggerInterface $logger,
-        VariableFactory $variable
+        VariableFactory $variable,
+        Dom $converter,
+        Config $config
     ) {
         $this->curl = $curl;
-        $this->storeManager = $storeManager;
+        $this->jsonSerializer = $jsonSerializer;
         $this->converter = $converter;
         $this->log = $logger;
         $this->variable = $variable;
-
-        $this->filePath = $this->getFeedUrl();
+        $this->config = $config;
     }
 
+    /**
+     * @throws CronException|NoSuchEntityException
+     */
+    public function processAggregateScores(): void
+    {
 
-    public function getFeedUrl(){
-        return $this->storeManager->getStore()->getConfig(
-            'kiyoh_settings/general/feed_url'
-        );
-    }
-
-    public function processAggregateScores() {
+        if (!$this->config->getLocationId()) {
+            throw new CronException(__('Location ID missing, please set your location ID.'));
+        }
 
-        $this->curl->get($this->filePath);
-        $output = $this->curl->getBody();
+        if (!$this->config->getApiKey()) {
+            throw new CronException(__('API key missing, please set your API key.'));
+        }
 
-        $doc = simplexml_load_string($output);
+        $this->curl->addHeader('X-Publication-Api-Token', $this->config->getApiKey());
+        $this->curl->get(config::PUBLICATION_URL . $this->config->getLocationId());
```

Comment:
Please use `Config::PUBLICATION_URL`

Reply:
What would also be nice is if you use a `sprintf()` here, like this:

```php
sprintf(Config::PUBLICATION_URL, $this->config->getLocationId());
```

That would require a change in `Model/Config.php` for this:

```php
const PUBLICATION_URL = 'https://www.kiyoh.com/v1/publication/review/external/location/statistics?locationId=%s';
```

The `%s` can also be a `%d` if the ID is always a numeric value.